lib/charms/opensearch/v0/opensearch_profile.py

```
@@ -0,0 +1,286 @@
+        return ClusterTopologyRequirements(
+            cluster_managers=3,
+            data=3,
+        )
+
+
+class TestingProfile(OpenSearchProfile):
+    """Testing profile for opensearch.
+
+    Ensures basic system requirements and 1 CM+ 1 Data roles.
+    """
+
+    type = PerformanceType.TESTING
+
+    @property
+    def memory_requirements(self) -> ProfileMemoryRequirements:
+        """Get the memory requirements for this profile."""
+        return ProfileMemoryRequirements(
+            memory_size=None,
+            jvm_heap_percentage=None,
+        )
+
+    @property
+    def cluster_topology_requirements(self) -> ClusterTopologyRequirements:
+        """Get the cluster topology requirements for this profile."""
+        return ClusterTopologyRequirements(
+            cluster_managers=1,
+            data=1,
+        )
+
+
+class ProfilesManager:
+    """Manage all profile related operations"""
+
+    def __init__(self, state: "OpenSearchClusterState", workload: "OpenSearchDistribution"):
+        self.state = state
+        self.workload = workload
+        self.profile = self.state.unit.profile or self.get_config_profile()
+        if self.profile.type == PerformanceType.TESTING:
+            logger.warning(
+                "Testing profile is used. This profile is not suitable for production use and should only be used for testing purposes."
+            )
+
+    def _apply_system_requirement(self, system_requirement: str, value: int) -> bool:
+        """Apply a system requirement."""
+        try:
+            self.workload._run_cmd(f"sysctl -w {system_requirement}={value}")
+            return int(self.workload._run_cmd(f"sysctl -n {system_requirement}")) == value
+        except OpenSearchCmdError:
+            return False
+
+    def _get_kernel_property_value(self, prop: str) -> int:
+        """Get the value of a kernel parameter."""
+        return int(self.workload._run_cmd(f"sysctl -n {prop}"))
```

We are not handling errors here, I will suggest to use a try except block not to put charm in unexpected error state.

---

I think it is ok to raise in this case. If we cannot execute a sysctl command on the unit then we should raise an error to try again.